keps/sig-cli/2775-kubectl-delete-interactivity-delay/README.md

```diff
@@ -0,0 +1,579 @@
+* https://docs.cloudbees.com/docs/cloudbees-codeship/latest/basic-builds-and-configuration/set-environment-variables
+* https://docs.drone.io/pipeline/environment/reference/ci/
+* https://devcenter.wercker.com/administration/environment-variables/available-env-vars/
+* https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-ci
+
+Do Not:
+* https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
+* https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values
+* https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
+* https://www.jetbrains.com/help/teamcity/predefined-build-parameters.html#Server+Build+Properties
+* https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-belowJenkinsSetEnvironmentVariables
+* https://jenkins-x.io/docs/resources/guides/using-jx/pipelines/envvars/
+* https://confluence.atlassian.com/bamboo/bamboo-variables-289277087.html
+* https://docs.cloudbees.com/docs/cloudbees-codeship/latest/pro-builds-and-configuration/environment-variables#_default_environment_variables (not on pro version?)
+* https://docs.gocd.org/current/faq/dev_use_current_revision_in_build.html#standard-gocd-environment-variables
+
+There is concern that these protections are not enough as they require users to opt-in for confirmation. This KEP is viewed as a starting point to introduce some form of protection that users may start to use. A way for users to set this as the default in the future will be in a future KEP.
+
+## Design Details
+
+If `--all` or `--all-namespaces` flags are passed in for a delete operation a warning will be printed and an artificial delay will be imposed. We need to determine what the correct length is for this. Initial thoughts were between 5 and 10 seconds. User testing will be done to gauge how long is needed to read the warning message. If `CI=true` is set in the environment this delay can be skipped.
+
+The warning message will be printed to stderr. We need to decide if we print a generic warning message or detail out the resources that will be deleted. The latter will be more complicated (especially calculating the resources deleted under a namespace) but more rewarding to the user. Deleting a namespace with `--dry-run=server` does not return any resources in the namespace.
+
+> Warning! You are deleting resources with `--all`. This will delete all of the (pods) in the (cluster or current namespace). Proceeding in x seconds. ctrl+c to interrupt
+
+> Warning! You are deleting the following resources:
+> Name:    Group:        Kind:
+> my-pod    core/v1      Pod
+> Proceeding in x seconds. ctrl+c to interrupt
+
+If the `--interactive` flag is passed for imperative deletes we will prompt the user for confirmation. As above we need to determine the prompt -- a generic message or explicitly lay out the resources.
+
+```
+kubectl delete namespace prod -i
+Are you sure? (y/n)
```

Comment:
suggest default to N.  e.g. ` Are you sure? (y/N)`

Reply:
Good call. Changed and explicitly called out.

Reply:
Is y/N is the standard across Kubernetes ? In some places, we do use Yes/No.

Reply:
I'll check for precedent.

Reply:
Just tried locally, how initial development looks like (pardon for the verbose/warning :) )
<img width="1016" alt="Screen Shot 2021-06-20 at 8 36 18 PM" src="https://user-images.githubusercontent.com/9973300/122696581-0841ca00-d209-11eb-9ad5-027869c342b6.png">

@thockin @liggitt @eddiezane - If the interactive flag is not given, it suppose to go for delay. we can delay it using ticker but every second is expect to log as a line like ( Ignore the Order ascending or descending )

->resources will be deleted in 9 sec
->resources will be deleted in 8 sec... ??

Not sure if it is possible to do like state changes/clock in node/UI  - resources will be delete in (9..8..7.....) ?